blueprints/data-solutions/dataproc-lifecycle-via-composer/terraform/main.tf

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+provider "google-beta" {
+  project = var.project_id
+  region  = var.region
+}
+
+locals {
+  shared_vpc_project = try(var.network_config.host_project, null)
+  use_shared_vpc     = var.network_config != null
+  subnet = (
+    local.use_shared_vpc
+    ? var.network_config.subnet_self_link
+    : values(module.vpc.0.subnet_self_links)[0]
+  )
+  vpc_self_link = (
+    local.use_shared_vpc
+    ? var.network_config.network_self_link
+    : module.vpc.0.self_link
+  )
+}
+
+module "vpc" {
+  source     = "../../../../modules/net-vpc"
+  count      = local.use_shared_vpc ? 0 : 1
+  project_id = var.project_id
+  name       = "vpc"
+  subnets = [
+    {
+      ip_cidr_range      = "10.128.0.0/20"
+      name               = "subnet"
+      region             = var.region
+      secondary_ip_range = {}
+    }
+  ]
+}
+
+resource "google_compute_firewall" "default-dataproc-allow-internal" {
+  project     = var.project_id
+  name        = "default-dataproc-allow-internal"
+  network     = "${replace(local.vpc_self_link,"https://www.googleapis.com/compute/v1/","")}"
+  description = "Allow internal traffic on the default network"
+  source_ranges = ["10.128.0.0/20"]
+  allow {
+    protocol = "udp"
+    ports    = ["0-65535"]
+  }
+  allow {
+    protocol = "icmp"
+  }
+  allow {
+    protocol  = "tcp"
+    ports     = ["0-65535"]
+  }
+  depends_on = [module.vpc]
+}
+
+resource "google_storage_bucket" "dataproc_jobs_bucket" {
+  project       = var.project_id
+  name          = "${var.region}-${var.project_id}-dataproc-jobs"
+  location      = var.region
+}
+
+resource "google_storage_bucket_object" "dataproc_job_files" {
+  for_each = fileset(path.module, "../jobs/*")
+  name   = "${replace(each.value,"../","")}"
+  source = each.value
+  bucket = "${replace(replace(google_storage_bucket.dataproc_jobs_bucket.name,"gs://",""),"/jobs","")}"
+}
+
+resource "google_service_account" "composer_env_service_account" {
+  project      = var.project_id
+  account_id   = "composer-env-service-account"
+  display_name = "Service Account for Composer Environment"
+}
+
+resource "google_project_iam_member" "composer_owner_binding" {
```

Comment:
Please use the relevant resource modules (in this case project module with `project_create = false`) to configure IAM, instead of low level resources.

Reply:
Done